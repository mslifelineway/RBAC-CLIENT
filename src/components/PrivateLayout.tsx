import React from "react";
import Box from "@mui/material/Box";
import CssBaseline from "@mui/material/CssBaseline";
import { MainContent } from "./MainContent";
import { StyledDrawerHeader } from "./Drawer/styled";
import { Drawer } from "./Drawer";
import { AppBar } from "./AppBar";

interface PrivateLayoutProps {
  children: React.ReactNode;
}

export const PrivateLayout = ({ children }: PrivateLayoutProps) => {
  const [open, setOpen] = React.useState(false);

  const handleDrawerOpen = () => {
    setOpen((prev) => !prev);
  };

  const handleDrawerClose = () => {
    setOpen(false);
  };
  return (
    <Box sx={{ display: "flex" }}>
      <CssBaseline />
      <AppBar handleDrawerOpen={handleDrawerOpen} />
      <Drawer open={open} handleDrawerClose={handleDrawerClose} />
      <Box component="main" sx={{ flexGrow: 1, p: 3 }}>
        <StyledDrawerHeader />
        <MainContent>{children}</MainContent>
      </Box>
    </Box>
  );
};
