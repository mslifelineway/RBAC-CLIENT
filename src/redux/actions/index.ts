export * from "./auth.action";
export * from "./account.action";
