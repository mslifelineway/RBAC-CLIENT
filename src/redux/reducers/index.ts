import authReducer from "./auth.slice";
import accountReducer from "./account.slice";
import roleReducer from "./role.slice";
import permissionReducer from "./permission.slice";

export const reducers = {
  authReducer,
  accountReducer,
  roleReducer,
  permissionReducer,
};
