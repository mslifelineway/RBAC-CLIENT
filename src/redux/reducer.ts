import { combineReducers } from "@reduxjs/toolkit";
import { reducers } from "./reducers";

export const rootReducer = combineReducers(reducers);
