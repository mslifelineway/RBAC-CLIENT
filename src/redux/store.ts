import { configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useSelector } from "react-redux";
import { rootReducer } from "./reducer";
import Cookies from "js-cookie";
import { CURRENT_USER } from "../constants";

const persistedState = Cookies.get(CURRENT_USER)
  ? JSON.parse(Cookies.get(CURRENT_USER)!)
  : null;

export const store = configureStore({
  reducer: rootReducer,
  preloadedState: {
    authReducer: {
      isAuthenticating: false,
      isAuthenticated: Boolean(persistedState),
      error: null,
      data: persistedState,
    },
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;

export const RequestStatus = {
  FULFILLED: "fulfilled",
  PENDING: "pending",
  REJECTED: "rejected",
};
