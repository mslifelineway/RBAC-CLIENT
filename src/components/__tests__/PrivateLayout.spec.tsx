import { create } from "react-test-renderer";
import { PrivateLayout } from "../PrivateLayout";
import { Provider } from "react-redux";
import { store } from "../../redux/store";

jest.mock("react-router-dom", () => {
  const original = jest.requireActual("react-router-dom");

  return {
    ...original,
    Link: ({ to, children }: { to: string; children: React.ReactNode }) => (
      <a href={to}>{children}</a>
    ),
  };
});

jest.mock("react-dom", () => {
  const original = jest.requireActual("react-dom");

  return {
    ...original,
    createPortal: (node: JSX.Element) => node,
  };
});

describe("PrivateLayout Component", () => {
  test("renders correctly with children", () => {
    const component = create(
      <Provider store={store}>
        <PrivateLayout>
          <div>Child Component</div>
        </PrivateLayout>
      </Provider>
    );
    expect(component.toJSON).toMatchSnapshot();
  });
});
