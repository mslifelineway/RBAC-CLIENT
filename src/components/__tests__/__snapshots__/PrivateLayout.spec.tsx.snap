// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateLayout Component renders correctly with children 1`] = `[Function]`;
